volatility3/framework/plugins/windows/mbrscan.py

```diff
@@ -0,0 +1,109 @@
+                                                                'PARTITION_TABLE': mbr.PARTITION_TABLE,
+                                                                'PARTITION_ENTRY': mbr.PARTITION_ENTRY
+                                                               })
+
+        partition_table_object = symbol_table + constants.BANG + "PARTITION_TABLE"
+        
+        # Define Signature and Data Length
+        mbr_signature = b"\x55\xAA"
+        mbr_length = 0x200
+        bootcode_length = 0x1B8
+
+        # Scan the Layer for Raw Master Boot Record (MBR) and parse the fields
+        for offset, _value in layer.scan(context = self.context, scanner = scanners.MultiStringScanner(patterns = [mbr_signature])):
+            try:
+                mbr_start_offset = offset - (mbr_length - len(mbr_signature))
+                partition_table = self.context.object(partition_table_object, offset = mbr_start_offset, layer_name = layer.name)
+
+                # Extract only BootCode
+                full_mbr = layer.read(mbr_start_offset, mbr_length, pad = True)
+                bootcode = full_mbr[:bootcode_length]
+                
+                if bootcode:
+                    all_zeros = bootcode.count(b"\x00") == len(bootcode)
+
+                if not all_zeros:
+                    partition_entries = [ 
+                                            partition_table.FirstEntry,
+                                            partition_table.SecondEntry,
+                                            partition_table.ThirdEntry,
+                                            partition_table.FourthEntry
+                                        ]
+                    partition_info = "\n"
+
+                    for index, partition_entry_object in enumerate(partition_entries):
+                        partition_entry_object.set_index(index)
+                        partition_info += str(partition_entry_object)
+                    
+                    yield 0, (
+                            format_hints.Hex(offset),
+                            partition_table.get_disk_signature(),
+                            self.get_hash(bootcode),
+                            self.get_hash(full_mbr),
+                            partition_info,
+                            interfaces.renderers.Disassembly(bootcode, 0, architecture),
+                            format_hints.HexBytes(bootcode)
+                    )
+                else:
+                    vollog.log(constants.LOGLEVEL_VV, f"Not a valid MBR: Data all zeroed out : {format_hints.Hex(offset)}")
+            
+            except exceptions.PagedInvalidAddressException:
+                pass
+
+    def run(self):
+        return renderers.TreeGrid([
+            ("Potential MBR at Physical Offset", format_hints.Hex),
+            ("Disk Signature", str),
+            ("Bootcode md5", str),
+            ("Bootcode (FULL) md5", str),
```

Comment:
This label seems inaccurate, since it's not the bootcode, but just the full MBR?